File: TDDMicroExercises/CSharp/TelemetrySystem.SomeDependencies/TelemetryDiagnosticControlsClient2.cs
using System;
namespace TDDMicroExercises.TelemetrySystem.SomeDependencies
{
    public class TelemetryDiagnosticControlsClient2
    {
		// A class with the only goal of simulating a dependency on TelemetryDiagnosticControls
		// that has impact on the refactoring.

		public TelemetryDiagnosticControlsClient2()
		{
			var teleDiagnostic = new TelemetryDiagnosticControls();

			teleDiagnostic.CheckTransmission();

			var result = teleDiagnostic.DiagnosticInfo;
		}
    }
}

File: TDDMicroExercises/Java/TelemetrySystem/src/main/java/tddmicroexercises/telemetrysystem/somedependencies/TelemetryDiagnosticControlsClient2.java
package tddmicroexercises.telemetrysystem.somedependencies;

import tddmicroexercises.telemetrysystem.TelemetryDiagnosticControls;

public class TelemetryDiagnosticControlsClient2
{
	// A class with the only goal of simulating a dependency on TelemetryDiagnosticControls
	// that has impact on the refactoring.

	public TelemetryDiagnosticControlsClient2() throws Exception {

		TelemetryDiagnosticControls teleDiagnostic = new TelemetryDiagnosticControls();

		teleDiagnostic.checkTransmission();

		String result = teleDiagnostic.getDiagnosticInfo();
	}
}
